lib/rucio/core/rse.py

```diff
@@ -445,18 +445,26 @@ def get_rse_attribute(key, rse_id=None, value=None, session=None):
 
     :returns: A list with RSE attribute values for a Key.
     """
-    rse_attrs = []
-    if rse_id:
-        query = session.query(models.RSEAttrAssociation.value).filter_by(rse_id=rse_id, key=key).distinct()
-        if value:
-            query = session.query(models.RSEAttrAssociation.value).filter_by(rse_id=rse_id, key=key, value=value).distinct()
-    else:
-        query = session.query(models.RSEAttrAssociation.value).filter_by(key=key).distinct()
-        if value:
-            query = session.query(models.RSEAttrAssociation.value).filter_by(key=key, value=value).distinct()
-    for attr_value in query:
-        rse_attrs.append(attr_value[0])
-    return rse_attrs
+
+    result = REGION.get('%s-%s-%s' % (key, rse_id, value))
```

Comment:
I am wondering if this is conflicting with respect to line 407.
Because there the cache is populated with (key, value, lookup_key) but a list of rses is stored. You store (key, rse_id, value) but the rse_attrs are stored. I wonder if there is a combination where the tuple is the same, which would then crash the procedure.
It might be better in both caches to prefix it with ('get_rse_attribute-%s-%s-%s').